logger/Logger.h

```diff
@@ -394,30 +403,35 @@ class Logger
 
 // Log line if the provided severity is below or equals the configured one
 #define FAIR_LOG(severity) \
-    for (bool fairLOggerunLikelyvariable = false; fair::Logger::Logging(fair::Severity::severity) && !fairLOggerunLikelyvariable; fairLOggerunLikelyvariable = true) \
-        fair::Logger(fair::Severity::severity, MSG_ORIGIN)
+    if (!fair::Logger::SuppressSeverity(fair::Severity::severity)) \
```

Comment:
There is a potential annoying problem with an `if` here.
If one uses this macro in an unfortunately-constructed if-else statement without braces, it will behave incorrectly.
Here is an example to demonstrate: https://wandbox.org/permlink/Pj7ZzXORljfiZ3t2
The macro will end up inserting an unexpected `if` and break the flow (when no braces are used).
This is the reason for the beautiful `for` loop that you see there - with the for loop there is no such unintentional insertion possible.

To solve this particular issue one could integrate your check in the existing for loop (in the condition part), or replace the `if` statement with another for loop. However I'm not sure if this will break the constexpr-ness...